extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -59,6 +67,25 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
+    if (source instanceof KafkaRecordEntity) {
```

Comment:
wdyt about making an overloaded constructor that's `@VisibleForTesting` that takes in protobufReaders (OpenCensus and OpenTelemetry) and then mocking them out in tests? the that way we won't need to maintain tests that verify the correctness of the OpenTelemetryProtobufReader in both this module and the druid-opentelemetry-extensions module. We could instead just verify that the correct protobuf reader is used when a header is sent.

Reply:
Agree, but it's still good to have the test because we do pass in a bunch of shared configs and we want those to behave the same.

Reply:
I tried to do the overload constructor but found this made the code , and testing extra complicated, considering how to construct the reader is more complicated than converter. So I still changed the code back to how it was. 

Besides, i made valueDemension can be configured and nullable for opentelemtryMetricsProtobufReader. 

This has been tested again locally with sandbox, and verified this can work well for ingestion and query.
![Screen Shot 2022-02-17 at 5 40 11 PM](https://user-images.githubusercontent.com/1749814/154601336-8133debd-c6d7-4ab5-913b-b83822e4d41e.png)
![Screen Shot 2022-02-17 at 5 40 23 PM](https://user-images.githubusercontent.com/1749814/154601342-987ad34b-29d8-454b-a39d-05bf97cd817b.png)